packages/xo-server/src/api/sr.js

```diff
@@ -64,7 +64,7 @@ export async function destroy({ sr }) {
   const config = xapi.xo.getData(sr, 'xosan_config')
   // we simply forget because the hosted disks are being destroyed with the VMs
   await xapi.forgetSr(sr._xapiId)
-  await asyncMapSettled(config.nodes, node => xapi.deleteVm(node.vm.id))
+  await asyncMapSettled(config.nodes, node => xapi.VM_destroy(node.vm.$ref))
```

Comment:
`node.vm` isn't an XAPI object. See https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/api/xosan.js#L658

Reply:
See https://github.com/vatesfr/xen-orchestra/blob/e40e35d30c1f3577f701237c43aa98f1d7cdd128/packages/xo-server/src/xapi-object-to-xo.js#L872-L874

```suggestion
  await asyncMapSettled(config.nodes, node => xapi.VM_destroy(node.vm._xapiRef))
```

Reply:
`node.vm`  only contains the `id` and the `ip` properties.

Reply:
Hmm, you're right, then `getXapiObject(node.vm.id).$destroy()`.

Reply:
Not better `xapi.VM_destroy(getXapiObject(node.vm.id).$ref)` ?

Reply:
Your version does the same thing, it's just longer.

Reply:
but `$destroy()` are calling `VM_destroy()` ?

Reply:
https://github.com/vatesfr/xen-orchestra/pull/5693#discussion_r601662796